File: 6-ZigzagConversion/6-ZigzagConversion.js
// Last updated: 30/07/2026, 10:09:24
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function(s, numRows) {
    if (numRows === 1) return s;

    let rows = Array(numRows).fill("");
    let i = 0;
    let step = 1;

    for (let char of s) {
        rows[i] += char;

        if (i === 0) step = 1;
        if (i === numRows - 1) step = -1;

        i += step;
    }

    return rows.join("");
};

File: 6-ZigzagConversion/6-ZigzagConversion.cpp
// Last updated: 30/07/2026, 10:09:19
class Solution {
public:
    string convert(string s, int numRows) {
        if (numRows == 1) return s;

        vector<string> v(numRows);
        int row = 0;
        bool down = true;

        for (char ch : s) {
            v[row] += ch;

            if (row == 0)
                down = true;
            else if (row == numRows - 1)
                down = false;

            if (down)
                row++;
            else
                row--;
        }

        string ans = "";
        for (string str : v)
            ans += str;

        return ans;
    }
};